File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/kit/KitClaimResult.java
package top.likoslupus.cellulosesz.api.kit;

import top.likoslupus.cellulosesz.api.text.LocalizedMessage;

import java.util.Map;

public record KitClaimResult(
        boolean success,
        LocalizedMessage message
) {

    public static KitClaimResult success(LocalizedMessage message) {
        return new KitClaimResult(true, message);
    }

    public static KitClaimResult success(String key) {
        return new KitClaimResult(true, LocalizedMessage.of(key));
    }

    public static KitClaimResult success(String key, Map<String, ?> placeholders) {
        return new KitClaimResult(true, LocalizedMessage.of(key, placeholders));
    }

    public static KitClaimResult failure(LocalizedMessage message) {
        return new KitClaimResult(false, message);
    }

    public static KitClaimResult failure(String key) {
        return new KitClaimResult(false, LocalizedMessage.of(key));
    }

    public static KitClaimResult failure(String key, Map<String, ?> placeholders) {
        return new KitClaimResult(false, LocalizedMessage.of(key, placeholders));
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/kit/KitClaimResult.kt
package top.likoslupus.cellulosesz.api.kit

import top.likoslupus.cellulosesz.api.text.LocalizedMessage
import top.likoslupus.cellulosesz.api.text.MessageArguments

@JvmRecord
public data class KitClaimResult(
    public val success: Boolean,
    public val message: LocalizedMessage
) {

    public companion object {

        @JvmStatic
        public fun success(message: LocalizedMessage): KitClaimResult =
            KitClaimResult(true, message)

        @JvmStatic
        public fun success(key: String): KitClaimResult =
            KitClaimResult(true, LocalizedMessage.of(key))

        @JvmStatic
        public fun success(key: String, arguments: MessageArguments): KitClaimResult =
            KitClaimResult(true, LocalizedMessage.of(key, arguments))

        @JvmStatic
        public fun failure(message: LocalizedMessage): KitClaimResult =
            KitClaimResult(false, message)

        @JvmStatic
        public fun failure(key: String): KitClaimResult =
            KitClaimResult(false, LocalizedMessage.of(key))

        @JvmStatic
        public fun failure(key: String, arguments: MessageArguments): KitClaimResult =
            KitClaimResult(false, LocalizedMessage.of(key, arguments))

    }

}
